mpReview.py

```diff
@@ -1579,39 +1579,21 @@ def onAcceptFiducialsPrompt(self):
         #name = splitted[1]
         label = slicer.util.getNode(splitted[2])
         try:
-          centroid = self.getCentroidForLabel(label, selectedID)
+          centroid = ModuleLogicMixin.getCentroidForLabel(label, int(selectedID))
           logging.debug("Creating fiducial at position %f, %f, %f" % tuple(centroid))
           addedFiducialIds.append(fiducialNode.AddFiducialFromArray(centroid, label.GetName()))
-        except:
-          notificationDialog = slicer.util.infoDisplay
-          notificationDialog("No label object with label %s. \n You might have forgotten to print a label."
-                             "To prevent the duplication of fiducials, all fiducials of the current "
-                             "creation step will be deleted." % label.GetName(), windowTitle='mpReview')
+        except Exception as exc:
+          message = "No label object with label %s. \n You might have forgotten to print a label. To prevent the " \
+                    "duplication of fiducials, all fiducials of the current creation step will be deleted. " \
+                    "For further information see details." % label.GetName()
+          slicer.util.errorDisplay(message, detailedText=str(exc.message), windowTitle='mpReview')
           self.removeFiducialIDsFromNode(fiducialNode, addedFiducialIds)
           return
 
   def removeFiducialIDsFromNode(self, node, ids):
     for idx in reversed(ids):
       node.RemoveMarkup(idx)
 
-  def getCentroidForLabel(self, label, labelId):
-    # TODO: take care about labels without anything drawn and labels with separate regions
-    ls = sitk.LabelShapeStatisticsImageFilter()
-    dstLabelAddress = sitkUtils.GetSlicerITKReadWriteAddress(label.GetName())
-    dstLabelImage = sitk.ReadImage(dstLabelAddress)
-    ls.Execute(dstLabelImage)
-    centroid = ls.GetCentroid(int(labelId))
-    IJKtoRAS = vtk.vtkMatrix4x4()
-    label.GetIJKToRASMatrix(IJKtoRAS)
-    order = label.ComputeScanOrderFromIJKToRAS(IJKtoRAS)
-    if order == 'IS':
-        centroid = [-centroid[0], -centroid[1], centroid[2]]
-    elif order == 'AP':
-        centroid = [-centroid[0], -centroid[2], -centroid[1]]
-    elif order == 'LR':
```

Comment:
Looking at this code now, I don't understand why these ifs are here. It should be sufficient to just flip L to R and P to S.